spec/i18n_spec.rb

```diff
@@ -107,6 +107,18 @@ def allowed_untranslated_key?(locale, key)
     expect(missing_interpolation_argument_keys).to be_empty
   end
 
+  it 'has matching HTML tags' do
+    i18n.data[i18n.base_locale].select_keys do |key, _node|
+      if key.start_with?('i18n.transliterate.rule.') || i18n.t(key).is_a?(Array) || i18n.t(key).nil?
+        next
+      end
+
+      html_unique_tags = i18n.locales.map { |locale| i18n.t(key, locale)&.scan(/<.+?>/) }.uniq
+
+      expect(html_unique_tags.size).to eq(1), "HTML tag mismatch for key #{key}"
```

Comment:
maybe we can change the structure of this or the message to indicate what the tags are?
```suggestion
      html_unique_tags = i18n.locales.map { |locale| i18n.t(key, locale)&.scan(/<.+?>/) }.uniq

      expect(html_unique_tags.size).to eq(1), "HTML tag mismatch for key: #{key}, tags: #{html_unique_tags}"
```

Reply:
Yeah, I actually added that locally to help troubleshoot the existing issues, so probably good to include _something_ similar, though hopefully moving forward it should be obvious enough to the person adding the new locale strings that there might be a problem with their tags.

As proposed, it wasn't especially helpful to debugging some of the more gnarly strings like [the one you mentioned](https://github.com/18F/identity-idp/pull/8847#discussion_r1272727736), since there's so many tags that it's hard to parse and make use of the output.

Reply:
Since I happen to have the history in my terminal, here's what it ended up looking like for that one 🙈 

```
  1) I18n has matching HTML tags
     Failure/Error: expect(html_tags.size).to eq(1), "HTML tag mismatch for key #{key}: #{html_tags}"
       HTML tag mismatch for key user_mailer.personal_key_sign_in.help_html: [["<p>", "</p>", "<p>", "</p>", "<p>", "<ol>", "<li>", "<strong>", "<a href=\"%{reset_password_url}\">", "</a>", "</strong>", "</li>", "<li>", "<strong>", "<a href=\"%{account_url}\">", "</a>", "</strong>", "</li>", "<li>", "<strong>", "<a href=\"%{account_url}\">", "</a>", "</strong>", "</li>", "</ol>", "</p>", "<br>", "<br>"], ["<p>", "</p>", "<p>", "</p>", "<p>", "<ol>", "<li>", "<strong>", "<a href=\"%{reset_password_url}\">", "</a>", "</strong>", "</li>", "<li>", "<strong>", "<a href=\"%{account_url}\">", "</a>", "</strong>", "</li>", "<li>", "<strong>", "</strong>", "</li>", "</ol>", "</p>", "<br>", "<br>"]]
```

Reply:
ah I see your point. I guess it's fine as-is and having the key name is enough to debug